drivers/serial/uart_opentitan.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2023 by Rivos Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/drivers/uart.h>
+#include <soc.h>
+#include <zephyr/irq.h>
+
+/* Register offsets within the UART device register space. */
+#define UART_INTR_STATE_REG_OFFSET 0x0
+#define UART_INTR_ENABLE_REG_OFFSET 0x4
+#define UART_CTRL_REG_OFFSET 0x10
+#define UART_STATUS_REG_OFFSET 0x14
+#define UART_RDATA_REG_OFFSET 0x18
+#define UART_WDATA_REG_OFFSET 0x1c
+#define UART_FIFO_CTRL_REG_OFFSET 0x20
+#define UART_OVRD_REG_OFFSET 0x28
+#define UART_TIMEOUT_CTRL_REG_OFFSET 0x30
+
+/* Control register bits. */
+#define UART_CTRL_TX_BIT 0
+#define UART_CTRL_RX_BIT 1
+#define UART_CTRL_NCO_OFFSET 16
+
+/* FIFO control register bits. */
+#define UART_FIFO_CTRL_RXRST_BIT 0
+#define UART_FIFO_CTRL_TXRST_BIT 1
+
+/* Status register bits. */
+#define UART_STATUS_TXFULL_BIT 0
+#define UART_STATUS_RXEMPTY_BIT 5
+
+#define DT_DRV_COMPAT lowrisc_opentitan_uart
+
+#define DEV_CFG(dev) \
+	((const struct uart_opentitan_config *const)(dev)->config)
```

Comment:
We are trying to move away from this macro. Not needed.

Reply:
Removed.